### PR TITLE
Rework - Requesting presentations

### LINKED ISSUE #397: Feature: Participants and Companies must be able to request presentations
The current edition doesn't allow to request presentations yet. There are a lot edge cases to this but the basic business logic goes as:
- A participant can request only one presentation

- A company can request only one presentation __UNLESS__
    - The company is a gold sponsor - they have 2 presentations
    - The company is HZ - they have unlimited presentations 

- A speaker can have only one presentation

### PR DESCRIPTION
**Pull request by v-stamenova**
# Description

In the current edition the feature to request presentations was not yet reworked to match the current database. This branch introduces the rework. 

NOTE: The policy for who can request presentation is from last year

closes #397, #398

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## What needs to be tested

- [x] You can request presentation as participant
- [x] You can request presentation as company rep
- [x] You can request presentation as company speaker
- [x] You cannot request 2 presentations
- [x] You can update the details of the presentation
- [x] You can see the details of the presentation
- [x] You can delete the presentation

# Checklist:

- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Livewire/Presentation/EditPresentationModal.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+     */
+    public function save() : RedirectResponse
+    {
+        $this->validate();
+
+        $this->form->update();
+
+        if (Auth::user()->presenter_of->id == $this->presentation->id) {
+            return redirect(route('presentations.show', $this->presentation))
+                ->with('status', 'Presentation successfully updated.');
```
Comment (TimKardol) [inline app/Livewire/Presentation/EditPresentationModal.php:41]: Cannot edit presentation details. Following error occurs:
`App\Livewire\Presentation\EditPresentationModal::save(): Return value must be of type Illuminate\Http\RedirectResponse, Livewire\Features\SupportRedirects\Redirector returned`

**Review by TimKardol — DISMISSED** [review]

**Comment by IGORnvk** [comment]
Снимок экрана 2024-04-27 124535
For some reason, there is an empty space on description field when I try to create the request. It doesn't appear on any other fields.

**Comment by IGORnvk** [comment]
Снимок экрана 2024-04-27 124700
Also in the edit modal, all the selected text is actually white and you can not see it because the background is also white.

**Comment by IGORnvk** [comment]
> Снимок экрана 2024-04-27 124700 Also in the edit modal, all the selected text is actually white and you can not see it because the background is also white.

Oh yeah, this is related to dark mode. Only cool guys use dark mode.

**Review by IGORnvk — APPROVED** [review]
Good job!

**Pull request merged**